content/Software Support/PLC IDE/If-PLC-IDE-isnt-working-with-your-device.md

```diff
@@ -0,0 +1,229 @@
+---
+title: "If PLC IDE isn't working with your device"
+id: 8824551178780
+---
+
+Learn what to do if you experience an issue using your Opta or Portenta Machine Control with PLC IDE.
+
+> ![TIP]
+> If you're just getting started with PLC IDE, see [Arduino® PLC IDE Setup & Device License Activation](https://docs.arduino.cc/software/plc-ide/tutorials/plc-ide-setup-license/).
+
+In this article:
+
+- [If you can't find the serial port for your device](#if-you-cant-find-the-serial-port-for-your-device)
+- [If there's an issue when you install the runtime](#if-theres-an-issue-when-you-install-the-runtime)
+- [If there's an issue when you connect to the device](#if-theres-an-issue-when-you-connect-to-the-device)
+- [If you see an error when programming the device](#if-you-see-an-error-when-programming-the-device)
+- [If there is an issue activating the license](#if-there-is-an-issue-activating-the-license)
+
+---
+
+## If you can't find the serial port for your device
+
+In this section:
+
+- [About serial ports in PLC IDE](#about-serial-ports-in-plc-ide)
+- [If you don't know which port to use](#if-you-dont-know-which-port-to-use)
+- [If your device does not appear on any port](#if-your-device-does-not-appear-on-any-port)
+
+### About serial ports in PLC IDE
+
+There are two contexts where you need to specify a serial port target.
+
+- _When installing the runtime._
+- _When configuring the Modbus protocol._
+
+When the runtime has been installed, two serial ports may appear. When installing the runtime, the choice of port doesn't matter. When configuring the Modbus protocol, you should select the **lower-numbered** port as the target.
+
+### If you don't know which port to use
+
+In some cases, it may be challenging to determine which serial port belongs to your PLC device. This is especially true if other devices or interfaces have ports open on your computer.
+
+1. If two ports are open, and they are numbered COM<N> and COM<N+1> (for example, COM20 and COM21), it's likely that both are created by your PLC device. Choose the **lower-numbered port** when configuring PLC IDE.
+2. If many ports are available, try disconnecting other USB devices that may be using serial communication and select **Project > Refresh current target** to refresh the list of ports.
```

Comment:
@marqdevx We tested it. To update the list of COM ports available in the “Modbus Config” dialog, it is sufficient to close the dialog and open it again (by clicking on **Properties** for the **Modbus** protocol).

But because this section refers to any context where the user doesn't know which serial port to choose (i.e. runtime installation and modbus config) , we want to suggest a solution that works for both.